src/precice/impl/ParticipantImpl.cpp

```diff
@@ -408,12 +398,6 @@ void ParticipantImpl::advance(
 
   advanceCouplingScheme();
 
-  if (_couplingScheme->isTimeWindowComplete()) {
-    for (auto &context : _accessor->readDataContexts()) {
-      context.moveToNextWindow();
-    }
-  }
```

Comment:
Also important regarding extrapolation: Previously we called `moveToNextWindow` for the storage of the `ReadDataContext` here. I moved this inside the coupling scheme, because then we don't have to worry about convergence etc. in `ParticipantImpl` anymore.